dev/tools/lib/repository.dart

```diff
@@ -84,31 +83,31 @@ abstract class Repository {
   /// If the repository will be used as an upstream for a test repo.
   final bool localUpstream;
 
-  Directory _checkoutDirectory;
+  Directory? _checkoutDirectory;
 
   /// Directory for the repository checkout.
   ///
   /// Since cloning a repository takes a long time, we do not ensure it is
   /// cloned on the filesystem until this getter is accessed.
   Directory get checkoutDirectory {
```

Comment:
You could use `late final` to set this up automagically, assuming some light wrapper

```dart
late final Directory checkoutDirectory = lazilyInitialize(parentDirectory.childDirectory(name));
```

You might need to tweak `lazilyInitialize` to return the original directory

Reply:
good point, i'll try that. i also need to fix the tests.

Reply:
Migrating to late final is breaking my test, I'll follow up with a PR isolating that change.